File: app/src/main/java/com/i2medier/financialpro/adapter/SimpleInterestAdapter.java
package com.i2medier.financialpro.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.i2medier.financialpro.R;
import com.i2medier.financialpro.model.MonthModel;
import com.i2medier.financialpro.util.Utils;

import java.util.ArrayList;


public class SimpleInterestAdapter extends RecyclerView.Adapter<SimpleInterestAdapter.ViewHolder> {
    Context context;
    ArrayList<MonthModel> monthModels;

    public SimpleInterestAdapter(Context context, ArrayList<MonthModel> arrayList) {
        this.context = context;
        this.monthModels = arrayList;
    }

    @Override
    @NonNull
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        return new ViewHolder(LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_simple_intrest, viewGroup, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        viewHolder.txtYear.setText(Utils.decimalFormat.format(this.monthModels.get(i).getYear()));
        viewHolder.txtInterest.setText(Utils.decimalFormat.format(this.monthModels.get(i).getInterest()));
        viewHolder.txtBalance.setText(Utils.decimalFormat.format(this.monthModels.get(i).getBalance()));
        if (i % 2 == 0) {
            viewHolder.llMain.setBackgroundColor(this.context.getResources().getColor(R.color.colorWhite));
        } else {
            viewHolder.llMain.setBackgroundColor(this.context.getResources().getColor(R.color.colorLight));
        }
    }

    @Override
    public int getItemCount() {
        return this.monthModels.size();
    }


    public class ViewHolder extends RecyclerView.ViewHolder {
        LinearLayout llMain;
        TextView txtBalance;
        TextView txtInterest;
        TextView txtYear;

        public ViewHolder(@NonNull View view) {
            super(view);
            this.llMain = (LinearLayout) view.findViewById(R.id.llMain);
            this.txtYear = (TextView) view.findViewById(R.id.txtYear);
            this.txtInterest = (TextView) view.findViewById(R.id.txtInterest);
            this.txtBalance = (TextView) view.findViewById(R.id.txtBalance);
        }
    }
}

File: app/src/main/java/com/i2medier/financialpro/adapter/SimpleInterestAdapter.kt
package com.i2medier.financialpro.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.i2medier.financialpro.R
import com.i2medier.financialpro.model.MonthModel
import com.i2medier.financialpro.util.Utils

class SimpleInterestAdapter(
    private val context: Context,
    private var monthModels: ArrayList<MonthModel>
) : RecyclerView.Adapter<SimpleInterestAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_simple_intrest, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val model = monthModels[position]
        holder.txtYear.text = Utils.decimalFormat.format(model.year)
        holder.txtInterest.text = Utils.decimalFormat.format(model.Interest)
        holder.txtBalance.text = Utils.decimalFormat.format(model.Balance)
        holder.llMain.setBackgroundColor(
            if (position % 2 == 0) ContextCompat.getColor(context, R.color.colorWhite)
            else ContextCompat.getColor(context, R.color.colorLight)
        )
    }

    override fun getItemCount(): Int = monthModels.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val llMain: LinearLayout = itemView.findViewById(R.id.llMain)
        val txtYear: TextView = itemView.findViewById(R.id.txtYear)
        val txtInterest: TextView = itemView.findViewById(R.id.txtInterest)
        val txtBalance: TextView = itemView.findViewById(R.id.txtBalance)
    }
}
